src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -86,61 +93,63 @@ export class RgwMultisiteWizardComponent extends BaseModal implements OnInit {
   }
 
   ngOnInit(): void {
-    this.open = this.route.outlet === 'modal';
-    this.rgwDaemonService
-      .list()
-      .pipe(
-        switchMap((daemons) => {
-          this.daemons = daemons;
-          const daemonStatsObservables = daemons.map((daemon) =>
-            this.rgwDaemonService.get(daemon.id).pipe(
-              map((daemonStats) => ({
-                hostname: daemon.server_hostname,
-                port: daemon.port,
-                frontendConfig: daemonStats['rgw_metadata']['frontend_config#0']
-              }))
-            )
-          );
-          return forkJoin(daemonStatsObservables);
-        })
-      )
-      .subscribe((daemonStatsArray) => {
-        this.rgwEndpoints.value = daemonStatsArray.map((daemonStats) => {
-          const protocol = daemonStats.frontendConfig.includes('ssl_port') ? 'https' : 'http';
-          return `${protocol}://${daemonStats.hostname}:${daemonStats.port}`;
-        });
-        const options: SelectOption[] = this.rgwEndpoints.value.map(
-          (endpoint: string) => new SelectOption(false, endpoint, '')
+    this.open = this.route.outlet === 'modal';  
```

Comment:
This is interesting, why are we managing this from here ? Why not like other modals in dashboard ?

Reply:
I see similar usage in a couple of more components as well, I don't have a definite answer for this. @nizamial09 Can you help here?

Reply:
This is usually done if the modal is routed. Routed modals are not opened by default so we are force opening it based on the route and then close it later when the route is changed. So if this is a routed modal, then this is correct.

Reply:
Okay noticed this is with carbonization,  we can create a our own modal component around `cds-modal` so we may avoid setting such very basic functionality.

Reply:
I will bring this in sync and then create tracker perhaps. Thanks